Gradle task runAntlrKey does not declare its dependencies correctly

## Description

The Gradle task of parser generation for the `KeYJavaParser` is not correctly set-up, especially, the imported grammars are not declared as inputs. 

This leads to the problem, that this task is not correctly re-executed. Leading to annyoing exceptions: 

```
* What went wrong:
Execution failed for task ':key.core:compileJava' (registered by plugin class 'org.gradle.api.plugins.JavaBasePlugin').
> Compilation failed; see the compiler output below.
  /home/weigl/work/key-main/key.core/src/main/java/de/uka/ilkd/key/nparser/builder/TacletPBuilder.java:196: error: cannot find symbol
          if (!ctx.IGNOREUPDATELEVEL().isEmpty() && !ctx.SAMEUPDATELEVEL().isEmpty()) {
                  ^
    symbol:   method IGNOREUPDATELEVEL()
    location: variable ctx of type TacletContext
  4 errors
```

Introduced by #3822 

## Reproducible
  
always

### Steps to reproduce 

1. Run `gradle testClasses`
1. Make a change in an imported grammar file. This needs to resolve into a compiler error.
2. Run `gradle testClasses`

   Observe that it still compiles. 


> What is your expected behavior and what was the actual behavior?

Compile error.

### Additional information

Branch: `main` 75f0a043fe843e936a86bf295d883ec1d5598a2e

---

* Commit: 75f0a043fe843e936a86bf295d883ec1d5598a2e

<!-- END OF BUG TEMPLATE -->

--- a/key.ncore/build.gradle
+++ b/key.ncore/build.gradle
@@ -20,7 +20,10 @@ def antlr4OutputKey = String.valueOf(projectDir) + '/build/generated-src/antlr/m
 
 tasks.register('runAntlr4Key', JavaExec) {
     //see incremental task api, prevents rerun if nothing has changed.
-    inputs.files "src/main/antlr/KeYLexer.g4", "src/main/antlr/KeYParser.g4"
+    inputs.files "src/main/antlr/KeYLexer.g4", "src/main/antlr/KeYParser.g4",
+            "src/main/antlr/KeYCommonParser.g4", "src/main/antlr/KeYGlobalDeclParser.g4",
+            "src/main/antlr/KeYSequentParser.g4", "src/main/antlr/KeYTacletParser.g4",
+            "src/main/antlr/KeYTermParser.g4"
     outputs.dir antlr4OutputKey
     classpath = configurations.antlr4
     mainClass.set("org.antlr.v4.Tool")

